receiver/splunkhecreceiver/receiver.go

```diff
@@ -391,9 +392,20 @@ func (r *splunkReceiver) handleRawReq(resp http.ResponseWriter, req *http.Reques
 	}
 }
 
-func (r *splunkReceiver) extractChannelHeader(req *http.Request) (string, bool) {
-	if headers, ok := req.Header[splunk.HTTPSplunkChannelHeader]; ok {
-		return headers[0], true
+func (r *splunkReceiver) extractChannel(req *http.Request) (string, bool) {
+	// check header
+	for k, v := range req.Header {
+		if strings.EqualFold(k, splunk.HTTPSplunkChannelHeader) {
+			return strings.ToUpper(v[0]), true
```

Comment:
Why do we need to convert the value to uppercase?

Reply:
this is to ensure the case insensitive nature of UUID. Do you think this would be better implemented on the ackextension side?

Reply:
Is it predefined in the eck extension that the channel must be UUID? It can be any string, then this logic should stay here

Reply:
Ah that's right. I will keep the logic here then.